applications/GeoMechanicsApplication/custom_strategies/strategies/geo_mechanics_newton_raphson_erosion_process_strategy.hpp

```diff
@@ -140,6 +181,19 @@ class GeoMechanicsNewtonRaphsonErosionProcessStrategy
             KRATOS_ERROR_IF_NOT(converged) << "Groundwater flow calculation failed to converge." << std::endl;
         }
 
+        if (mStoreNonconvergedSolutionsFlag) {
+            CollectCurrentSolution();
+            auto& r_dof_set = this->GetBuilderAndSolver()->GetDofSet();
+            mNonconvergedSolutionsMatrix = Matrix( r_dof_set.size(), mNonconvergedSolutions.size() );
+            for (std::size_t i = 0; i < mNonconvergedSolutions.size(); ++i) {
+                block_for_each(r_dof_set, [&](const auto& r_dof) {
```

Comment:
Could you explicitly capture the necessary variables for the lambda function? I think this should work:
```suggestion
                block_for_each(r_dof_set, [this, &mNonconvergedSolutions, i](const auto& r_dof) {
```

Reply:
probably this will give problems with the member, since we can use c++14 I would change to `[&rNonConvergedSolutions = this->mNonconvergedSolutions, i]` or just capture `this` but without referencing `mNonconvergedSolutions` in the capture clause.